tidy3d/plugins/autograd/invdes/parametrizations.py

```diff
@@ -1,46 +1,124 @@
 
 from ..types import KernelType, PaddingType
-from .filters import make_filter
+from .filters import AbstractFilter, CircularFilter, ConicFilter
 from .projections import tanh_projection
 
 
+class FilterAndProject(Tidy3dBaseModel):
+    """A class that combines filtering and projection operations.
+
+    Parameters
+    ----------
+    filter : AbstractFilter
+        The filter to apply.
+    beta : float = 1.0
+        The beta parameter for the tanh projection.
+    eta : float = 0.5
+        The eta parameter for the tanh projection.
+    """
+
+    filter: AbstractFilter
+    beta: pd.NonNegativeFloat = 1.0
+    eta: pd.NonNegativeFloat = 0.5
+
+    def __call__(self, array: NDArray, beta: float = None, eta: float = None) -> NDArray:
+        """Apply the filter and projection to an input array.
+
+        Parameters
+        ----------
+        array : NDArray
+            The input array to filter and project.
+        beta : float = None
+            The beta parameter for the tanh projection. If None, uses the instance's beta.
+        eta : float = None
+            The eta parameter for the tanh projection. If None, uses the instance's eta.
+
+        Returns
+        -------
+        NDArray
+            The filtered and projected array.
+        """
+        filtered = self.filter(array)
+        beta = beta if beta is not None else self.beta
+        eta = eta if eta is not None else self.eta
+        projected = tanh_projection(filtered, beta, eta)
+        return projected
+
+    @classmethod
+    def from_params(
+        cls,
+        radius: Union[float, Tuple[float, ...]] = None,
+        dl: Union[float, Tuple[float, ...]] = None,
+        *,
+        size_px: Union[int, Tuple[int, ...]] = None,
+        beta: float = 1.0,
+        eta: float = 0.5,
+        filter_type: KernelType = "conic",
+        padding: PaddingType = "reflect",
+    ) -> FilterAndProject:
+        """Create a FilterAndProject instance from parameters.
+
+        Parameters
+        ----------
+        radius : Union[float, Tuple[float, ...]] = None
+            The radius of the kernel. Can be a scalar or a tuple.
+        dl : Union[float, Tuple[float, ...]] = None
+            The grid spacing. Can be a scalar or a tuple.
+        size_px : Union[int, Tuple[int, ...]] = None
+            The size of the kernel in pixels for each dimension. Can be a scalar or a tuple.
+        beta : float = 1.0
+            The steepness of the tanh projection. Higher values result in a sharper transition.
+        eta : float = 0.5
+            The midpoint of the tanh projection.
+        filter_type : KernelType = "conic"
+            The type of filter kernel to use.
+        padding : PaddingType = "reflect"
+            The padding type to use for the filter.
+
+        Returns
+        -------
+        FilterAndProject
+            An instance of FilterAndProject.
+        """
+        if filter_type == "conic":
```

Comment:
seems like duplicated code? maybe this should get written as a function or class method somewhere?